src/ref_serializer.rs

```diff
@@ -40,32 +41,29 @@ pub struct RefParameterSerializer<'a> {
 }
 
 impl RefParameterSerializer<'_> {
-    pub fn parameter(self, name: &str, value: impl AsRefBareItem) -> SFVResult<Self> {
-        Serializer::serialize_parameter(name, value, self.buffer)?;
-        Ok(self)
+    pub fn parameter<'b>(self, name: &KeyRef, value: impl Into<RefBareItem<'b>>) -> Self {
+        Serializer::serialize_parameter(name, value, self.buffer);
+        self
     }
 }
 
 /// Serializes `List` field value components incrementally.
 /// ```
-/// use sfv::{RefBareItem, RefListSerializer};
+/// use sfv::{key_ref, string_ref, token_ref, RefListSerializer};
 ///
 /// let mut serialized_item = String::new();
 /// let serializer = RefListSerializer::new(&mut serialized_item);
 /// serializer
 ///     .bare_item(11)
-///     .unwrap()
-///     .parameter("foo", true)
+///     .parameter(key_ref("foo"), true)
```

Comment:
Let's use KeyRef::from_str here. We don't want people to just use the panicking versions immediately.
We can give examples with the panicking ones too, as long as they're not the first ones 🙂